### PR TITLE
Serverless Implemention of The TTKD App

### PR DESCRIPTION
**Pull request by lkretvix**

### CODE REVIEW

File: TTKDTrainingApp/App.js [hunk 1]
```diff
@@ -1,17 +1,55 @@
...
+    ReactCamera: ReactCamera,
+    VideoWithControls: VideoWithControls,
+    Content: Content,
+  },
+  {
+    initialRouteName: 'HomeScreen',
+    headerLayoutPreset: 'center',
+    defaultNavigationOptions: {
+      headerStyle: {
+        backgroundColor: 'gray',
```
Comment (angbamfield) [inline TTKDTrainingApp/App.js]: nippick/question: is styling supposed to go in a css file or naw ?

File: TTKDTrainingApp/src/redux/reducers/reactCamera.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import {ADD_RECORDED_VIDEO} from '../actionTypes';
+
+const initialState = {
+  recordedVideos: [],
```
Comment (angbamfield) [inline TTKDTrainingApp/src/redux/reducers/reactCamera.js]: is this where the URIs would technically be ?

File: TTKDTrainingApp/src/components/HomeScreen.js [hunk 3]
```diff
@@ -1,77 +1,58 @@
...
   );
-
-  const renderTitle = (
-    <View style={styles.title}>
-      <Text style={styles.titleText}>
-        {categoryId ? category.category_name : 'TTKD Home'}
-      </Text>
-    </View>
+  const subCategories = useSelector(state =>
+    getSubCategories(state, categoryId),
```
Comment (angbamfield) [inline TTKDTrainingApp/src/components/HomeScreen.js]: do we have use cases when there are subcategories that have subcategories? Or the use case is just main category and then only one level down of categories can exist ?

File: TTKDTrainingApp/App.js (hunk 1 above)
Comment (lkretvix) [reply TTKDTrainingApp/App.js]: In the docs FaceBook doesn't specify that it should be in a separate file, and many examples use styles in the same file so I went that way. It has the added benefit of keeping the file structure slightly cleaner without having Components and their Style files inside folders

File: TTKDTrainingApp/src/redux/reducers/reactCamera.js (hunk 2 above)
Comment (lkretvix) [reply TTKDTrainingApp/src/redux/reducers/reactCamera.js]: This is a list of video objects that are the videos recorded by the user. The objects contain this structure:
```
{
"contentId", 
"deviceOrientation", 
"isRecordingInterrupted", 
"uri", 
"videoOrientation"
}
```
Everything here except for `contentId `is provided by `react-native-camera` when recording a video, and `contentId` is used to filter the videos when displaying them on a Content screen

File: TTKDTrainingApp/src/components/HomeScreen.js (hunk 3 above)
Comment (lkretvix) [reply TTKDTrainingApp/src/components/HomeScreen.js]: Subcategories can exist here. When someone taps a `CategoryCard` that is a subCategory, we push the `HomeScreen` component onto the navigation stack and set the current category to the selected `categoryId` of the pressed `CategoryCard`

Comment (lkretvix) [reply TTKDTrainingApp/src/components/HomeScreen.js]: I realized as I was typing ^^ that it may be a bit hard to follow, so here's an example:
category 1 is the main category, category 2 has 1 as its parent, and category 3 has 2 as its parent, so:
- `HomeScreen` -> categoryId is 1, shows category 2 as a subcategory
- Tap category 2, push `HomeScreen` on the stack with category 2
- `HomeScreen` categoryId is 2, shows category 3 as subcategory
- Tap category 3, push `HomeScreen` on the stack with category 3

File: TTKDTrainingApp/src/components/ReactCamera.js [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+          ref={cameraRef}
+          style={styles.preview}
+          type={
+            frontCamera
+              ? RNCamera.Constants.Type.front
+              : RNCamera.Constants.Type.back
+          }
+          flashMode={RNCamera.Constants.FlashMode.off}
+          captureAudio={false}
+          androidCameraPermissionOptions={{
```
Comment (angbamfield) [inline TTKDTrainingApp/src/components/ReactCamera.js]: ios asked me to do this automatically btw.... but you probably knew that anyways ahah!

File: TTKDTrainingApp/src/components/VideoWithControls/VideoWithControls.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+  const [paused, setPaused] = useState(true);
+  const [progress, setProgress] = useState(0);
+  const [duration, setDuration] = useState(0);
+  const [rate, setRate] = useState(DEFAULT_SPEED);
+  const [loading, setLoading] = useState(true);
+
+  ///////////////////// Effects /////////////////////
+
+  useEffect(() => {
+    if (!props.isFocused) {
```
Comment (angbamfield) [inline TTKDTrainingApp/src/components/VideoWithControls/VideoWithControls.js]: what is this isFocused prop ? Are we actually able to determine if a video is focused or not ?

File: TTKDTrainingApp/src/components/HomeScreen.js (hunk 3 above)
Comment (angbamfield) [reply TTKDTrainingApp/src/components/HomeScreen.js]: ah i see!

File: TTKDTrainingApp/src/components/VideoWithControls/VideoWithControls.js (hunk 5 above)
Comment (lkretvix) [reply TTKDTrainingApp/src/components/VideoWithControls/VideoWithControls.js]: `isFocused` is provided by the `react-native-navigation` library. it signifies if that component is focused, aka (in this instance) if we're on that screen or not. If we aren't, then pause the video

https://reactnavigation.org/docs/en/with-navigation-focus.html

File: TTKDTrainingApp/src/components/ReactCamera.js (hunk 4 above)
Comment (lkretvix) [reply TTKDTrainingApp/src/components/ReactCamera.js]: `react-native-camera` had some really good boiler plate code that had all the permissions stuff we needed luckily
